Content.Server/GameTicking/Rules/Components/NukeopsRuleComponent.cs

```diff
@@ -233,3 +233,10 @@ public enum WinCondition : byte
     SomeNukiesAlive,
     AllNukiesAlive
 }
+
+public enum NukieType
+{
+    Commander,
+    Agent,
+    Operative
+}
```

Comment:
i really dislike this, this is information you should be able to get from the entity at any moment. At least move the enum to a datafield in `NukeopsRoleComponent` and set it up when the role is added or something or infer it in some other way

Reply:
This isnt about knowing who is which role.
Its used by the spawning system to designate a team composition - previously it was just 
'If index is 0 then commander, if index is 1 then agent, else operative'

NukieType is sent to the spawning method along with the players session - where it is used to choose the name, gear and set the PrototypeId on the mind's NukeopsRoleComponent

Reply:
I mean thats kind of the issue isnt it? The best way to handle it is changing nukies to spawn the entities first and then assign the roles

Reply:
I mean thats kind of the problem isnt it? The best way to handle things like this is spawning the nukies entities first and then assign the roles i think

Reply:
I'm not following what you mean.

This is the only method that uses NukieType (other than OnPlayersSpawning, which assigns it to specific sessions)
https://github.com/space-wizards/space-station-14/blob/32beffe2ef099f6a2305389154ec2867d9ec8e9a/Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs#L802-L828

Its never stored in a component or used after spawning is completed - just a clearer version of what already exists with it hardcoding specific indexes in the list of sessions to roles:
https://github.com/space-wizards/space-station-14/blob/8f1d67014cc3c261ffba31226c6d6f89b0ce3d3d/Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs#L951-L953
https://github.com/space-wizards/space-station-14/blob/8f1d67014cc3c261ffba31226c6d6f89b0ce3d3d/Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs#L874-L901

It also allows for SpawnOperatives to be order agnostic - ie it doesnt need to be ordered Commander>Agent>Operatives.

Reply:
Can't you just make the data encapsulating it a struct? So name + role + gear, then just make a separate one for each nukie type and not tie it to an enum / switch statement.

Plus that way it's reuseable for other stuff eventually, e.g. ninja spawning.

Reply:
Removed the enums.
There are now NukeopSpawnPreset's defined in the component that contain the spawn details.